wagtailio/roadmap/models.py

```diff
@@ -0,0 +1,185 @@
+
+    content_panels = Page.content_panels + [
+        FieldPanel("sponsorship_page"),
+        FieldPanel("fine_print"),
+        FieldPanel("body"),
+    ]
+
+    promote_panels = Page.promote_panels + SocialMediaMixin.panels
+
+    search_fields = Page.search_fields + [
+        index.SearchField("body"),
+    ]
+
+    def get_context(self, request, *args, **kwargs):
+        context = super().get_context(request, *args, **kwargs)
+        items = Item.objects.filter(publish=True).order_by("sort_order")
+        context.update(
+            {
+                "milestones": Milestone.objects.filter(publish=True)
+                .order_by(models.F("due_on").asc(nulls_last=True))
+                .prefetch_related(models.Prefetch("items", queryset=items))
+            }
+        )
+        return context
+
+
+class Item(Orderable):
+    NEEDS_SPONSORSHIP_LABEL = "needs sponsorship"
+
+    publish = models.BooleanField(default=True)
+    needs_sponsorship = models.BooleanField(default=False, editable=False)
+    sponsorship_url = models.URLField(
+        blank=True,
+        verbose_name="Sponsorship URL",
+        help_text="Custom URL to use for the 'Sponsor this' label",
+    )
+    number = models.IntegerField(
+        unique=True,
+        editable=False,
+        help_text="GitHub issue number",
+    )
+    state = models.CharField(choices=State.choices, max_length=32, editable=False)
+    title = models.CharField(max_length=255)
+    url = models.URLField(verbose_name="URL")
+    milestone = ParentalKey(
+        "roadmap.Milestone",
+        on_delete=models.CASCADE,
+        related_name="items",
+    )
+    # For simplicity, store labels as a comma-separated text
+    labels = models.TextField(help_text="Comma-separated list of labels", blank=True)
+
+    panels = [
+        FieldPanel("publish", help_text=publish_help),
+        FieldPanel("sponsorship_url"),
+        MultiFieldPanel(
+            [
+                FieldPanel("title", widget=readonly),
+                FieldPanel("url", widget=readonly),
+                FieldPanel("labels", widget=readonly),
+            ],
+            heading="GitHub data",
+            help_text=github_help,
+        ),
+    ]
+
+    class Meta:
+        verbose_name = _("roadmap item")
+        verbose_name_plural = _("roadmap items")
+        ordering = ["sort_order", "number"]
+
+    def __str__(self):
+        return self.title
+
+    def clean(self):
+        self.labels = ",".join(self.labels_set)
+
+    @cached_property
+    def labels_set(self):
+        return set(self.labels.split(",")) - {""}
+
+    @cached_property
+    def labels_list(self):
+        return sorted(self.labels_set - {self.NEEDS_SPONSORSHIP_LABEL})
+
+    @cached_property
+    def needs_sponsorship(self):
+        return self.NEEDS_SPONSORSHIP_LABEL in self.labels_set
+
+
+class Milestone(ClusterableModel):
+    publish = models.BooleanField(default=True)
```

Comment:
I don’t understand what this `publish` is for either – in what scenario would we be using this to override the GitHub data? If there’s one then I think it’d be worth a comment, or have the help text explain the scenario.